Shipping Labels: Update default item details for customs form

itsmeichigo opened the pull request:
Part of #4687

# Description
This PR update default item details form custom form based on fetched products and variations. Item details should now display correct data.

# Changes
- Updated ShippingLabelsCustomsForm quantity property type to Decimal to match with order item's for safety.
- Customs form list view model: 
   - Fetch currency from order details, then send the currency all the way down to item detail form view models to display on the UI.
   - Reuse `ShippingLabelPackageDetailsResultsControllers` to fetch products and variations.
   - Observe changes of products and variations to update default item details. 
- Updated item details view model to fetch weight unit from `shippingSettingsService` to display on the UI.

# Demo
![item-details](https://user-images.githubusercontent.com/5533851/129357344-741f935c-5202-4435-a521-3aa617c1be06.gif)

# Testing steps
1. Make sure that your test store has WCShip plugin installed and package details set up.
2. Navigate to Orders tab and select an order with international shipping (origin country different from destination).
3. Select Create Shipping Label.
4. Skip through Ship To, Ship From, Package Details to configure Customs.
5. Notice that in Package Details section, item details are displayed correctly: description, weight, value, weight unit, currency. Origin country is set the same as your site address country by default.

Update release notes:
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

itsmeichigo committed:
Send currency to item detail form

itsmeichigo committed:
Display weight unit for item details

itsmeichigo committed:
Update item details based on fetched products and variations

itsmeichigo committed:
Remove didSet for customs form item publisher

itsmeichigo committed:
Use Site address country code as default origin country for customs form items

itsmeichigo committed:
Only update default item details

itsmeichigo committed:
Update customs form item quantity to Decimal to match order item and for safety

itsmeichigo committed:
Update default customs form item check to consider weight instead of item description

itsmeichigo committed:
Add unit test for customs form list view model

itsmeichigo committed:
Update unit test to check for default customs form items

itsmeichigo committed:
Display currency symbol instead of currency code on item details form

rachelmcr commented on WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Customs/ShippingLabelCustomsFormListViewModel.swift:
```diff
@@ -49,6 +61,101 @@ final class ShippingLabelCustomsFormListViewModel: ObservableObject {
+
+    /// Return copy of customs forms with updated item details based on fetched products and variations.
+    ///
+    func updateCustomsForms(products: [Product], productVariations: [ProductVariation]) -> [ShippingLabelCustomsForm] {
```
At some point here we should be sure to filter out any order items that are virtual (virtual products or virtual product variations), since those aren't shipped. Right now, those are showing up in the Package Content list for the customs form even though they aren't shown elsewhere in the shipping label flow.

itsmeichigo replied:
Thank you Rachel for pointing out this - I totally missed it earlier. I have fixed it in 6439bd453cf2249f46a72c78ca368cc5a675a656.

rachelmcr reviewed (commented):
This looks good and works great in most cases! The only issue I came across is with virtual products, which should be excluded (since they aren't shipped) but are showing up in the Package Content list on the customs form screen.

itsmeichigo committed:
Exclude virtual items from customs form item list

itsmeichigo committed:
Add unit test to check that virtual products are excluded from customs form item list

itsmeichigo commented:
Thank you @rachelmcr for pointing out the issue with virtual products - I have fixed it in 6439bd453cf2249f46a72c78ca368cc5a675a656 and added a unit test to check it. Please take another look and let me know what you think 🙇‍♀️

itsmeichigo committed:
Add unit test to check that nonexistent products are excluded from customs item list

rachelmcr reviewed (approved):
Code & testing looks great, thank you! ✅
